plugins/pluginmanagers/lazy.nix

```
@@ -1,223 +1,206 @@
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
-                  opts should be a table (will be merged with parent specs),
-                  return a table (replaces parent specs) or should change a table.
-                  The table will be passed to the Plugin.config() function.
-                  Setting this value will imply Plugin.config()
-                '';
-            };
-          });
+  extraConfig =
+    cfg:
+    let
+      lazyPlugins = cfg.plugins;
 
-          listOfPlugins = types.listOf pluginType;
+      lazyPath =
+        let
+          processPlugin =
+            plugin:
+            let
+              mkEntryFromDrv =
+                p:
+                if lib.isDerivation p then
+                  {
+                    name = "${lib.getName p}";
+                    path = p;
+                  }
+                else
+                  {
+                    name = "${lib.getName p.pkg}";
+                    path = p.pkg;
+                  };
+              processDependencies =
+                if plugin ? dependencies && plugin.dependencies != null then
+                  builtins.concatMap processPlugin plugin.dependencies
+                else
+                  [ ];
+            in
+            [ (mkEntryFromDrv plugin) ] ++ processDependencies;
+
+          processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
         in
-        mkOption {
-          type = listOfPlugins;
-          default = [ ];
-          description = "List of plugins";
-        };
-    };
-  };
-
-  config = mkIf cfg.enable {
-    extraPlugins = [ pkgs.vimPlugins.lazy-nvim ];
-
-    extraPackages = [ cfg.gitPackage ];
-
-    extraConfigLua =
-      let
-        pluginToLua =
-          plugin:
-          let
-            keyExists = keyToCheck: attrSet: lib.elem keyToCheck (lib.attrNames attrSet);
-          in
-          if isDerivation plugin then
-            { dir = "${lazyPath}/${lib.getName plugin}"; }
-          else
-            {
-              "__unkeyed" = plugin.name;
-
-              inherit (plugin)
-                cmd
-                cond
-                config
-                dev
-                enabled
-                event
-                ft
-                init
-                keys
-                lazy
-                main
-                module
-                name
-                optional
-                opts
-                priority
-                submodules
-                ;
-
-              dependencies = helpers.ifNonNull' plugin.dependencies (
-                if isList plugin.dependencies then (pluginListToLua plugin.dependencies) else plugin.dependencies
-              );
-
-              dir =
-                if plugin ? dir && plugin.dir != null then plugin.dir else "${lazyPath}/${lib.getName plugin.pkg}";
-            };
+        pkgs.linkFarm "lazy-plugins" processedPlugins;
```

```suggestion
        pkgs.linkFarmFromDrvs "lazy-plugins" (builtins.concatMap processPlugin lazyPlugins);
```

1. Use `linkFarmFromDrvs` to simplify `mkEntryFromDrv`
2. Inlining `processedPlugins` here is more readable IMO
3. Use `concatMap` instead of `concatLists` + `map`